### PR TITLE
TGR-109 Update Ruby version + GHA

### PR DESCRIPTION
**Pull request by croyfish**
## JIRA tickets:
- [SCSB Item Updater - update Ruby version + GHA](https://newyorkpubliclibrary.atlassian.net/browse/TGR-109)

## Things Done:
- Updated OS / Ruby / Rails / Gems
- Implemented GitHub Actions for PR builds
- Implemented GitHub Actions for deployments
- Set `qa` as the default branch and added branch protection rules to `qa` and `production`

## How to Test:

- Build the app locally via README instructions.
- Run the unit tests
- Check out the services running in the qa environment (nypl-digital-dev)

### CODE REVIEW

File: .github/workflows/build-production.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: scsb_item_updater
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
```
Comment (EdwinGuzman) [inline .github/workflows/build-production.yml:37]: I think it's fine if you push the image once with just `production-latest` as the tag, unless if there is a reason to push two similar images but with different tags?

File: .github/workflows/build-qa.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Build, tag, and push image to Amazon ECR
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: scsb_item_updater
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          docker tag $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG $ECR_REGISTRY/$ECR_REPOSITORY:qa-remediation
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:qa-remediation
```
Comment (EdwinGuzman) [inline .github/workflows/build-qa.yml:39]: Is this for testing and will later be updated to `qa-latest`?

**Review by EdwinGuzman — APPROVED** [review]
This installs, runs, and tests pass. What's not too clear is how to run the tests in the docker container and there's no section on tests in the readme. Still, running tests outside of the docker build/container pass for me.

File: .github/workflows/build-qa.yml (hunk 2 above)
Comment (croyfish) [reply .github/workflows/build-qa.yml:39]: Correct. Steven recommended this approach.

**Pull request merged**